test/common/http/header_utility_test.cc

```diff
@@ -414,6 +416,25 @@ invert_match: true
   EXPECT_FALSE(HeaderUtility::matchHeaders(unmatching_headers, header_data));
 }
 
+TEST(MatchHeadersTest, HeaderExactMatchIgnoreCase) {
+  TestRequestHeaderMapImpl matching_headers_1{{"match-header", "match-value"}};
+  TestRequestHeaderMapImpl matching_headers_2{{"match-header", "MATCH-VALUE"}};
+  TestRequestHeaderMapImpl unmatching_headers{{"match-header", "other-value"},
+                                              {"other-header", "match-value"}};
+  const std::string yaml = R"EOF(
+name: match-header
+exact_match: match-value
```

Comment:
Also can have a test case for 'Match-Value'

Reply:
done.